### PR TITLE
Improved stopping_condition method, and more performant wheel builds.

### PR DESCRIPTION
**Pull request by AmintorDusko**
**Context:** This PR optimizes the wheels build for Linux, improves the stopping_condition method for `lightning.gpu`, and improve the testing.

**Description of the Change:**
- Wheels for Linux are now parallelized and are taking less than 10 min.
- A new and improved `stopping_condition`, relying on the supports_operation method, was implemented. [[Lightning-#386]](https://github.com/PennyLaneAI/pennylane-lightning/pull/386)
- Tests were expanded to cover tape execution.
- Condition to run Expval tests was updated.

**Benefits:**
Faster wheel builds, more reliable `stopping_condition`, and more tests in the repository.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Comment by AmintorDusko** [comment]
I removed, for now, the workflow update_dev_version. Without the proper setting, it is guaranteed that it will always fail.

**Comment by AmintorDusko** [comment]
[sc-29886]

File: tests/test_expval.py [hunk 1]
```diff
@@ -22,13 +22,13 @@
 from conftest import U, U2, A
 
 try:
-    from pennylane_cuquantum.cuquantum_gpu import CPP_BINARY_AVAILABLE
+    from pennylane_lightning_gpu.lightning_gpu import CPP_BINARY_AVAILABLE
```
Comment (mlxd) [inline tests/test_expval.py:25]: :eyes:

**Review by mlxd — APPROVED** [review]
Great job @AmintorDusko 
Nothing extra to add here.

**Review by multiphaseCFD — APPROVED** [review]
Hey @AmintorDusko , Nice work!

**Pull request merged**